MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/activity/login/LoginActivity.kt

```diff
@@ -0,0 +1,167 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.macrobenchmark.target.activity.login
+
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Button
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.material.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.semantics.testTagsAsResourceId
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.dp
+import com.example.macrobenchmark.target.activity.MainActivity
+import com.example.macrobenchmark.target.util.SampleViewModel
+import com.example.macrobenchmark.target.util.TAG
+
+@OptIn(ExperimentalComposeUiApi::class)
+class LoginActivity : ComponentActivity() {
+
+    private val sampleViewModel by viewModels<SampleViewModel>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (intent.hasExtra("user") && intent.hasExtra("password")) {
+            Log.d(TAG, "onCreate: Found benchmark category")
+            intent.extras?.run {
+                Log.d(TAG, "onCreate: Using benchmark userdata")
+                val userName = getString("user", "")
+                val password = getString("password", "")
+                sampleViewModel.login(userName, password)
+            }
+        } else {
+            sampleViewModel.loadAppLogin()
+        }
+        setContent {
+            LoginScreen()
+        }
+    }
+
+    @Composable
+    fun LoginScreen() {
+        fullyDrawnReporter.addReporter()
```

Comment:
this is not the right approach - we want to remove the reporter when this composable is disposed.
Just use `ReportDrawnWhen { true }`

I'm thinking whether we should have an implementation without the predicate to just call it 🤔 when this is composed -- `ReportDrawn()`

Reply:
According to the docs we are able to add a reporter and remove it later on. 
IMO this is a viable approach to using this API.